Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1

```diff
@@ -426,4 +428,51 @@ class InventoryUtil {
             $global:VerbosePreference = $savedVerbosePreference
         }
     }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Resource pool location
+    #>
+    [PSObject] GetResourcePoolParent($resourcePoolLocation){
+        $locationItems = $resourcePoolLocation.Split('/')
```

Comment:
You can add here additional checks for empty location as well as location not containing **`/`**. In the first case, you should return the root folder for **`ResourcePool Inventory Items`** and in the second, call **`Get-Inventory`** with the **`$resourcePoolLocation`** parameter. 

For reference, you can check how **`GetDatacenter`** function is implemented here in the **`InventoryUtil`**.

Reply:
If the location is empty string, I'll throw an exception. As there's no single root resource pool. There's a 'Resources' RP for each host outside cluster and each cluster. 
If there is no '/' symbol in the location the code already does what you suggest. Splitting by non-existing symbol results one element array, containing the same string:
`$a = "NoDash"
$b = $a.Split('/')
$b[0]
NoDash`
Please correct me if I miss something.